src/arr/trove/statistics.arr

```diff
@@ -180,49 +190,65 @@ fun mode-any(l :: List<Number>) -> Number:
     ms.get(num-random(ms.length()))
   end
 end
-  
-fun stdev(l :: List) -> Number:
-  doc: ```returns the standard deviation of the list 
+
+fun variance(l :: List) -> Number:
+  doc: ```returns the variance of the list
        of numbers, or raises an error if the list is empty```
   reg-mean = mean(l)
   sq-diff = l.map(lam(k): num-expt((k - reg-mean), 2) end)
   sq-mean = mean(sq-diff)
-  num-sqrt(sq-mean)
+  sq-mean
 end
 
-fun stdev-sample(l :: List) -> Number:
-  doc: ```returns the standard deviation of the list 
+fun stdev(l :: List) -> Number:
+  doc: ```returns the standard deviation of the list
+       of numbers, or raises an error if the list is empty```
+  num-sqrt(variance(l))
+end
+
+fun variance-sample(l :: List) -> Number:
+  doc: ```returns the variance of the list
        of numbers, or raises an error if the list is empty```
   len = l.length()
   reg-mean = mean(l)
   sq-diff = l.map(lam(k): num-expt((k - reg-mean), 2) end)
   sq-mean = math.sum(sq-diff) / (len - 1)
-  num-sqrt(sq-mean)
+  sq-mean
+end
+
+fun stdev-sample(l :: List) -> Number:
+  doc: ```returns the standard deviation of the list
+       of numbers, or raises an error if the list is empty```
+  num-sqrt(variance-sample(l))
 end
 
-fun linear-regression(x :: List<Number>, y :: List<Number>) -> (Number -> Number):
-  doc: "returns a linear predictor function calculated with ordinary least squares regression"
-  if x.length() <> y.length():
+# please see: https://online.stat.psu.edu/stat462/
+
+fun multiple-regression(x_s_s :: List<List<Number>>, y_s :: List<Number>) -> (Any -> Number):
+  doc: "multiple-regression"
+  intercepts = MX.col-matrix.make(raw-array-of(1, y_s.length()))
+  mx_X = intercepts.augment(MX.lists-to-matrix(x_s_s))
+  mx_Y = MX.lists-to-matrix(y_s.map(lam(y): [list: y] end))
```

Comment:
This would be cheaper as `col-matrix.make(raw-array-of-list(y_s))`

Reply:
Done². Thanks²!